quark/core/shurikenapkinfo.py

```diff
@@ -0,0 +1,532 @@
+            "first_hex": firstResult["hex"],
+            "second": secondResult["smali"],
+            "second_hex": secondResult["hex"],
+        }
+
+    def __getClasses(self) -> Iterator[hdvmclass_t]:
+        match self.ret_type:
+            case "APK":
+                dexList = (
+                    self.analysis.get_dex_file_by_index(i)
+                    for i in range(self.analysis.get_number_of_dex_files())
+                )
+
+                rawClasses = (
+                    self.analysis.get_hdvmclass_from_dex_by_index(dex, i)
+                    for dex in dexList
+                    for i in range(
+                        self.analysis.get_number_of_classes_for_dex_file(dex)
+                    )
+                )
+
+            case "DEX":
+                rawClasses = (
+                    self.analysis.get_class_by_id(i)
+                    for i in range(self.analysis.get_number_of_classes())
+                )
+
+            case _:
+                raise ValueError("Unsupported File type.")
+        return rawClasses
+
+    @property
+    def superclass_relationships(self) -> Dict[str, Set[str]]:
+        rawClasses = self.__getClasses()
+
+        hierarchyDict = defaultdict(set)
+
+        for rawClass in rawClasses:
+            className = self._convertClassNameFormat(
+                rawClass.class_name.decode()
+            )
+            superclassName = self._convertClassNameFormat(
+                rawClass.super_class.decode()
+            )
+
+            hierarchyDict[className].add(superclassName)
+
+        return hierarchyDict
+
+    @property
+    def subclass_relationships(self) -> Dict[str, Set[str]]:
+        rawClasses = self.__getClasses()
+
+        hierarchyDict = defaultdict(set)
+
+        for rawClass in rawClasses:
+            className = self._convertClassNameFormat(
+                rawClass.class_name.decode()
+            )
+            superclassName = self._convertClassNameFormat(
+                rawClass.super_class.decode()
+            )
+
+            hierarchyDict[superclassName].add(className)
+
+        return hierarchyDict
+
+    def _convert_to_method_object(
+        self,
+        methodAnalysis: hdvmmethodanalysis_t,
+    ) -> MethodObject:
+        className = self._convertClassNameFormat(
+            methodAnalysis.class_name.decode()
+        )
+
+        return MethodObject(
+            # access_flags=methodAnalysis.access_flags,
+            class_name=className,
+            name=methodAnalysis.name.decode(),
+            descriptor=descriptor_to_androguard_format(
+                methodAnalysis.descriptor.decode()
+            ),
+            cache=methodAnalysis,
+        )
+
+    def _convertClassNameFormat(self, className: str) -> str:
+
+        if not className.endswith(";"):
+            className = "L" + className.replace(".", "/") + ";"
+
+        return className
+
+    def _convertMemberFieldFormat(self, memberField: str) -> str:
```

Comment:
If `memberField` is `android.support.v4.app.FragmentManagerImpl->mTmpActions java.lang.Runnable[]`, the result will be incorrect.